### PR TITLE
Documentation updates for Toolchain changes

### PR DESCRIPTION
**Pull request by SenRamakri**
Documentation updates for Toolchain changes planned for 5.12 release

### CODE REVIEW

**Comment by SenRamakri** [comment]
Please review - But **DO NOT MERGE** this until the compiler changes are done. We will update the comments in the PR once we are ready to merge this after tool chain work is done. I took a shortcut as of now to capture both IAR and ARMC changes in one PR. Its possible that one might happen but other might not, in that case we will update this PR accordingly. This covers - IOTCORE-825 and IOTCORE-709.

**Comment by deepikabhavnani** [comment]
Iar doc update PR was created long back - https://github.com/ARMmbed/mbed-os-5-docs/pull/885

In case one is updated can use that

**Review by kegilbert — APPROVED** [review]

**Review by AnotherButler — APPROVED** [review]

**Comment by SenRamakri** [comment]
@AnotherButler @kegilbert - I added more changes to this PR as we changed our requirements around ARMC6 transition and that caused more documentation changes. Please review.

**Review by kegilbert — APPROVED** [review]

**Comment by AnotherButler** [comment]
Does this still have a code dependency?

**Comment by AnotherButler** [comment]
@SenRamakri Is this ready to merge, or are we waiting on a code dependency?

**Comment by SenRamakri** [comment]
> @SenRamakri Is this ready to merge, or are we waiting on a code dependency?

@AnotherButler - Unfortunately we need to wait, as IAR8 just merged today and we have to start on ARMC6 tomorrow. Sorry that we have to keep you waiting on this.

**Comment by AnotherButler** [comment]
Could you please add a link here to the code PR?

**Comment by SenRamakri** [comment]
@AnotherButler - Yes, actually we don't have a specific PR, we need the branch feature-armc6 merged before we can merge this documentation change. In fact, the merge PR is still not generated and Im waiting on this PR now - https://github.com/ARMmbed/mbed-os/pull/9860, sorry for the convoluted info.
I'll update this PR once we are ready, Im assuming it might be end of tomorrow.

File: docs/reference/configuration/mbed_targets.md
Comment (bulislaw) [inline docs/reference/configuration/mbed_targets.md:197]: We should be explicit that any new targets should support ARMC6 not 5.

**Comment by deepikabhavnani** [comment]
Needs a rebase to resolve conflicts

File: docs/reference/configuration/mbed_targets.md
Comment (SenRamakri) [reply docs/reference/configuration/mbed_targets.md:197]: @bulislaw - Please see the updates.

**Comment by SenRamakri** [comment]
Rebased

**Review by deepikabhavnani — APPROVED** [review]

**Comment by AnotherButler** [comment]
@bulislaw Have all of your concerns been addressed?

**Comment by 0xc0170** [comment]
Master received ARMC6 update. We are close to 5.12rc1, this documentation should be part of it.

**Comment by SenRamakri** [comment]
@AnotherButler - ARMC6/IAR8 changes are merged to master. I think we are good to merge these changes now.

**Comment by AnotherButler** [comment]
@bulislaw I'm going to merge this later today. If you have further concerns, please create a new PR.

**Review by bulislaw — APPROVED** [review]

**Pull request merged**
